services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/_scheduler_base.py

```diff
@@ -676,32 +531,20 @@ async def _process_completed_tasks(
         iteration: Iteration,
         comp_run: CompRunsAtDB,
     ) -> None:
-        ...
-
-    @staticmethod
-    def _build_exclusive_lock_key(*args, **kwargs) -> str:
-        assert args  # nosec
-        return f"{kwargs['user_id']}:{kwargs['project_id']}:{kwargs['iteration']}"
-
-    @staticmethod
-    def _redis_client_getter(*args, **kwargs) -> RedisClientSDK:
-        assert kwargs  # nosec
-        zelf = args[0]
-        assert isinstance(zelf, BaseCompScheduler)  # nosec
-        return zelf.redis_client
-
-    @exclusive(
-        redis=_redis_client_getter,
-        lock_key=_build_exclusive_lock_key,
-    )
-    async def _schedule_pipeline(
+        """process tasks from the 3rd party backend"""
+
+    async def apply(
         self,
         *,
         user_id: UserID,
         project_id: ProjectID,
-        iteration: PositiveInt,
-        wake_up_callback: Callable[[], None],
+        iteration: Iteration,
     ) -> None:
+        """schedules a pipeline for a given user, project and iteration.
+
+        Arguments:
+            wake_up_callback -- a callback function that is called in a separate thread everytime a pipeline node is completed
```

Comment:
I don't think this `wake_up_callback` doc belongs here. Or am I misunderstanding something?